PWGHF/Core/HFMLResponse.h

```diff
@@ -0,0 +1,214 @@
+{
+  uint8_t tag{0};
+  if (track.hasTPC() && track.hasTOF()) {
+    SETBIT(tag, PIDStatus::TPCAndTOF);
+  } else if (track.hasTPC() && !track.hasTOF()) {
+    SETBIT(tag, PIDStatus::TPCOnly);
+  } else {
+    SETBIT(tag, PIDStatus::TOFOnly);
+  }
+  return tag;
+}
+
+/// Combine TPC and TOF nSigma
+/// \param nSigTPC is the number of sigmas for TPC
+/// \param nSigTOF is the number of sigmas for TOF
+/// \param tagPID is the tag on PID status of the track
+/// \return combined nSigma value
+template <typename T1, typename T2>
+T1 getCombinedNSigma(const T1& nSigTPC, const T1& nSigTOF, const T2& tagPID)
+{
+  if (TESTBIT(tagPID, PIDStatus::TPCAndTOF)) {
+    return std::sqrt((nSigTPC * nSigTPC + nSigTOF * nSigTOF) / 2);
+  }
+  if (TESTBIT(tagPID, PIDStatus::TPCOnly)) {
+    return std::abs(nSigTPC);
+  }
+
+  if (TESTBIT(tagPID, PIDStatus::TOFOnly)) {
+    return std::abs(nSigTOF);
+  }
+}
+
+template <typename T = float>
+class HFMLResponse
+{
+ public:
+  /// Default constructor
+  HFMLResponse() = default;
+  /// Default destructor
+  virtual ~HFMLResponse() = default;
+
+  /// Access model from CCDB
+  /// \param onnxFiles is a vector of onnx file names, one for each bin
+  /// \param ccdbApi is the CCDB API
+  /// \param pathCCDB is the model path in CCDB
+  /// \param timestampCCDB is the CCDB timestamp
+  void accessModelFromCCDB(const std::vector<std::string>& onnxFiles, o2::ccdb::CcdbApi& ccdbApi, std::string pathCCDB, long timestampCCDB)
+  {
+    uint8_t counterModel{0};
+    for (const auto& onnxFile : onnxFiles) {
+      std::map<std::string, std::string> metadata;
+      bool retrieveSuccess = ccdbApi.retrieveBlob(pathCCDB, ".", metadata, timestampCCDB, false, onnxFile);
+      if (retrieveSuccess) {
+        mPaths[counterModel] = onnxFile;
+      } else {
+        LOG(fatal) << "Error encountered while accessing the ML model from CCDB! Maybe the ML model doesn't exist yet for this runnumber/timestamp?";
+      }
+      ++counterModel;
+    }
+  }
+
+  /// Initialize class instance (import configurables and OnnxModels)
+  /// \param binsLimits is a vector containing bins limits
+  /// \param cuts is a LabeledArray containing selections per bin
+  /// \param cutDir is a vector telling whether to reject score values greater or smaller than the threshold
+  /// \param paths is a vector of onnx model paths
+  /// \param enableOptimizations is a switch no enable optimizations
+  /// \param threads
+  void init(const std::vector<double>& binsLimits, const o2::framework::LabeledArray<double>& cuts, const std::vector<int>& cutDir,
+            const std::vector<std::string>& paths, bool enableOptimizations, int threads = 0)
+  {
+    // import configurables
+    mBinsLimits = binsLimits;
+    mCuts = cuts;
+    mCutDir = cutDir;
+
+    mNetworks = std::vector<o2::ml::OnnxModel>(mNModels);
+    mPaths = std::vector<std::string>(mNModels);
+    mNModels = binsLimits.size() - 1;
+    mNClasses = (cutDir.size() >= 3) ? cutDir.size() : 2;
```

Comment:
I would avoid to infer the number of classes from this, because the risk is to always get `3` from the default value of the `hf_cuts_ml`. It would be better in my opinion to have an explicit argument, set by the user.

Reply:
I am putting a cofigurable to choose the number of classes. But I had made tests and, if I initialize the `HFMLResponse` object inside the `void init(InitContext const&) ...` function, then it is the JSON value that is selected.

Reply:
The JSON value is the one of the configurable, did I get properly?

Reply:
Yes, that's it :)